app/src/test/java/com/codesoom/assignment/controllers/ProductControllerTest.java

```diff
@@ -0,0 +1,117 @@
+package com.codesoom.assignment.controllers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import com.codesoom.assignment.application.ProductService;
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.exception.ProductNonExistException;
+
+class ProductControllerTest {
+
+    ProductController productController;
+    ProductService productService;
+
+    @BeforeEach
+    void setUp() {
+        productService = mock(ProductService.class);
+        productController = new ProductController(productService);
+    }
+
+    @DisplayName("장난감 리스트에 장난감이 아무것도 없을 때, 비어있는 것을 보장하는 테스트입니다.")
+    @Test
+    void getProductListFromEmptyListTest() {
+        List<Product> productList = new ArrayList<>();
+        given(productService.getProducts()).willReturn(productList);
+        assertThat(productController.getProducts()).isEmpty();
+    }
```

Comment:
테스트 코드는 하나의 과학 실험과 같습니다. 무엇을 테스트하는지, 어떤 조건에서 테스트하는지, 테스트 결과가 무엇이 되어야 하는지를 "쉬운" 언어로 간결하게 설명하면 동욱님 본인도 깜짝 놀랄 정도로 잘 읽히는 좋은 테스트 코드가 나올 거에요.

이 테스트의 설명문을 다음과 같이 업그레이드할 수 있을 것 같습니다. 다음은 제가 작성한 메시지입니다. 이 메시지를 읽어보시고 어떤 점이 다른지 열심히 생각해 보세요.


```suggestion
    @DisplayName("장난감 서비스에 등록된 상품이 없다면, 컨트롤러의 getProducts 메소드는 비어 있는 리스트를 리턴합니다.")
    @Test
    void getProductListFromEmptyListTest() {
        List<Product> productList = new ArrayList<>();
        given(productService.getProducts()).willReturn(productList);
        assertThat(productController.getProducts()).isEmpty();
    }
```

Reply:
> 어떤 조건에서 테스트하는지, 테스트 결과가 무엇이 되어야 하는지를 "쉬운" 언어로 간결하게 설명하면

이 부분을 명심하겠습니다.

추가적으로 리스트를 서비스로 추상화 시킬수도 있군요.

Reply:
> 하나의 테스트 코드는 하나의 과학 실험과 같습니다.
이 부분을 명심하여 ~~ 하였을 때 (조건) ~~를 반환합니다. (결과) 이러한 형식으로 작성하도록 해보겠습니다.